### PR TITLE
Survey cycle selection

### PR DESCRIPTION
**Pull request by minotogna**

### CODE REVIEW

File: webapp/loggedin/appHeader/components/cycleSelector.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  const cycleItems = cycleKeys.map(key =>
+    ({ key, value: `${i18n.t(`common.cycle`)} ${Number(key) + 1}` })
+  )
+  const cycleSelection = R.find(R.propEq('key', surveyCycle), cycleItems)
+
+  return (
+    <Dropdown
+      items={cycleItems}
+      selection={cycleSelection}
```
Comment (fefrain) [inline webapp/loggedin/appHeader/components/cycleSelector.js]: you can use Dropdonw itemKeyProp="key" and pass surveyCycle directly to selection

File: server/modules/auth/api/authApi.js [hunk 2]
```diff
@@ -1,60 +1,46 @@
+const Log = require('../../../log/log').getLogger('AuthAPI')
```
[the remark this answers is not in the document]
Comment (minotogna) [reply server/modules/auth/api/authApi.js]: alright, changed

File: server/modules/survey/manager/surveyManager.js [hunk 3]
```diff
@@ -179,19 +175,14 @@ const publishSurveyProps = async (surveyId, langsDeleted, client = db) => {
...
-    if (userPrefSurveyId === id)
-      await UserRepository.deleteUserPref(user, userPrefNames.survey, t)
-
-    await SurveyRepository.dropSurveySchema(id, t)
-    await SurveyRdbManager.dropSchema(id, t)
-
-    await SurveyRepository.deleteSurvey(id, t)
-  })
-}
+const deleteSurvey = async surveyId => await db.tx(async t => {
```
[the remark this answers is not in the document]
Comment (minotogna) [reply server/modules/survey/manager/surveyManager.js]: yes and no.. but I removed them

File: webapp/commonComponents/form/dropdown.scss [hunk 4]
```diff
@@ -18,7 +18,7 @@
   }
 
   .form-input {
-    border-bottom: none;
+    border-bottom: none !important;
```
[the remark this answers is not in the document]
Comment (minotogna) [reply webapp/commonComponents/form/dropdown.scss]: fixed

File: webapp/loggedin/appHeader/components/cycleSelector.js (hunk 1 above)
Comment (minotogna) [reply webapp/loggedin/appHeader/components/cycleSelector.js]: not really. surveyCycle is just the key, not the item itself. I have renamed the variable to surveyCycleKey for better clarity

**Pull request merged**